R/mail.R

```diff
@@ -6,110 +6,249 @@
-    splits <- stringi::stri_split_fixed(i, "/")
-    destination[[i]] <- paste0(splits[[1]][[length(splits[[1]])]])
+    # Extract the name without the .txt.gz extension
+    base_name <- gsub("\\.txt\\.gz$", "", link)
 
-    #download file and place it at the destination
-    httr::GET(i, httr::write_disk(destination[[i]], overwrite=TRUE))
-  }
+    # Parse the date from the base name
+    date_parsed <- as.Date(paste0("01-", base_name), format = "%d-%Y-%B")
+    if (is.na(date_parsed)) {
+      warning("Could not parse date from link: ", link)
+      next
+    }
+    year_month_clean <- format(date_parsed, "%Y%m")
+
+    # Download URL
+    txt_url <- paste0(mailing_list, gsub("\\.gz$", "", link))
+    gz_url <- paste0(mailing_list, link)
+
+    # Attempt to download the .txt file first
+    download_url <- txt_url
+    response <- httr::GET(download_url, httr::timeout(60))
+
+    if (httr::status_code(response) != 200) {
+      # Fallback to .gz file if .txt is unavailable
+      download_url <- gz_url
+      response <- httr::GET(download_url, httr::timeout(60))
+      if (httr::status_code(response) != 200) {
+        cat("Both .txt and .gz downloads failed for link: ", link, "\n")
+        next
+      }
+    }
 
-  #Return filenames
-  return(destination)
+    # Define the destination file
+    dest <- file.path(save_folder_path, paste0('kaiaulu_', year_month_clean, '.mbox'))
+
+    # Print diagnostic info
+    cat("Downloading: ", download_url, "\n")
+    cat("Saving to: ", dest, "\n")
+
+    # Write file to disk
+    if (grepl("\\.gz$", download_url)) {
+      # Download the .gz file
+      gz_file_path <- file.path(save_folder_path, paste0('kaiaulu_', year_month_clean, '.mbox.gz'))
+      httr::GET(download_url, httr::write_disk(gz_file_path, overwrite = TRUE), httr::timeout(60))
+
+      # Unzip the file
+      gz_con <- gzfile(gz_file_path, open = "rb")
+      out_con <- file(dest, open = "wb")
+      while (TRUE) {
+        bytes <- readBin(gz_con, what = raw(), n = 1024 * 1024)
+        if (length(bytes) == 0) break
+        writeBin(bytes, out_con)
+      }
+      close(gz_con)
+      close(out_con)
+
+      # Remove the gz file after unzipping
+      file.remove(gz_file_path)
+    } else {
+      # Download the .txt file directly
+      httr::GET(download_url, httr::write_disk(dest, overwrite = TRUE), httr::timeout(60))
+    }
+
+    # Add the downloaded file to the list
+    downloaded_files <- c(downloaded_files, dest)
+  }
 
+  # Return downloaded files
+  return(downloaded_files)
 }
 
 
-#' Convert pipermail archive files (.txt and .txt.gz) into an mbox format for use with \code{\link{parse_mbox}}
-#' @param filelist A vector of pipermail archive files from \code{\link{download_pipermail}}
-#' @return Returns `output`, the name of the resulting .mbox file in the current working directory
+#' Refresh mbox files downloaded via pipermail
+#' Uses the adopted file name convention by \code{\link{download_pipermail}} to identify
```

Comment:
Add one empty line here with `#'` to separate title and description. You may also want to double check the actual documentation of the function in Rstudio after running build (on osx cmd + shift + b) to see it is formatted poperly when the user call the function docs `?refresh_pipermail`.